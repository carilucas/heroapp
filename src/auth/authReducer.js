
import { LOGIN, LOGOUT } from './types';


export const authReducer = (state = {}, {type,payload}) => {
   switch (type) {
      case LOGIN:
        return payload
   
      case LOGOUT:
        return {logged:false}
   
      default:
         return state
   }
}
