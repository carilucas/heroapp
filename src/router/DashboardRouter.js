import React from 'react';
import {
   Route,
   Redirect,
   Switch
 } from "react-router-dom";
import { Navbar } from '../components/NavComponent';
import { DcComponent } from '../components/DcComponent';
import { MarvelComponent } from '../components/MarvelComponent';
import { SearchComponent } from '../components/SearchComponent';
import { HeroScreen } from '../components/HeroScreen';


export const DashboardRouter = () => {

   
   return (
      <div>
         <Navbar/>
         <div className="container mt-4">
               <Switch>
                  <Route exact path="/marvel" component={MarvelComponent}/>
                  <Route exact path="/dc" component={DcComponent}/>
                  <Route exact path="/search/" component={SearchComponent}/>
                  <Route exact path="/hero/:heroId" component={HeroScreen} />
                  <Redirect to='/marvel' />
               </Switch>
         </div>
      </div>
   )
}
