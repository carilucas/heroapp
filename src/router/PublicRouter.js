import React from 'react'
import { Redirect, Route } from 'react-router'
import { LoginComponent } from '../components/LoginComponent'

export const PublicRouter = ({
   logged,
   component:Component,
   ...rest
}) => {
   return (
      <Route 
         {...rest}
         component = {
            (props)=>(
               !logged
               
                  ? <LoginComponent {...props} /> 
                  : <Redirect to="/" />
            )
         }
      
      />
   )
}
