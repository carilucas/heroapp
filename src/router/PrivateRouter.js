import React from 'react'
import { Redirect, Route } from 'react-router'
import { DashboardRouter } from './DashboardRouter';
import PropTypes from 'prop-types'



export const PrivateRouter = ({
   logged,
   component:Component,
   ...rest
}) => {
   localStorage.setItem('lastPath',rest.location.pathname)
   return (
      <Route 
         {...rest}
         component = {
            (props)=>(
               logged
                  ? <DashboardRouter {...props} /> 
                  : <Redirect to="/login" />
            )
         }
      
      />
   )
}

PrivateRouter.propTypes = {
   logged : PropTypes.bool.isRequired,
   component : PropTypes.func.isRequired,
}