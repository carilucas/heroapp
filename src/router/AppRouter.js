import React, { useContext } from 'react'
import {
   BrowserRouter as Router,
   Switch
 } from "react-router-dom";
import { AuthCotext } from '../auth/authContext';
import { LoginComponent } from '../components/LoginComponent';
import { DashboardRouter } from './DashboardRouter';
import { PrivateRouter } from './PrivateRouter';
import { PublicRouter } from './PublicRouter';

export const AppRouter = () => {

  const {user} = useContext(AuthCotext);
  const {logged} = user;
  
   return (
      <Router>
      <div>
        <Switch>
            <PublicRouter exact path="/login" component={LoginComponent} logged={logged}/>
            <PrivateRouter path="/" component={DashboardRouter} logged={logged}/>
        </Switch>
      </div>
    </Router>
   )
}
