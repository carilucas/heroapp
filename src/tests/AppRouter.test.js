import { AppRouter } from "../router/AppRouter";
import {mount} from 'enzyme'
import { AuthCotext } from "../auth/authContext";

const dispatch = jest.fn()
describe('Pruebas en <AppRouter>', () => {
   

   test('Debe mostrar login si no está autenticado', () => {
      
      const user = {logged:false}

      const wrapper = mount(
      <AuthCotext.Provider value={{user,dispatch}}>
         <AppRouter/>
      </AuthCotext.Provider>
      
      );
      
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('h1').text()).toBe('LoginComponent');

   });

   test('Debe mostrar el componente marvel si está autenticado', () => {
      
      const user = {
         name: 'Fernando',
         logged:true
      }

      const wrapper = mount(
      <AuthCotext.Provider value={{user,dispatch}}>
         <AppRouter/>
      </AuthCotext.Provider>
      
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('.navbar').exists()).toBe(true);

   })
   
   
})
