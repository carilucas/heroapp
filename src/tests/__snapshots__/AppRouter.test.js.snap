// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter> Debe mostrar el componente marvel si est√° autenticado 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Hero App
        </a>
        <div
          className="navbar-collapse"
        >
          <div
            className="navbar-nav"
          >
            <a
              aria-current="page"
              className="nav-item nav-link active"
              href="/marvel"
              onClick={[Function]}
              style={Object {}}
            >
              Marvel
            </a>
            <a
              aria-current={null}
              className="nav-item nav-link"
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
            <a
              aria-current={null}
              className="nav-item nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </div>
        </div>
        <div
          className="navbar-collapse collapse w-100 order-3 dual-collapse2"
        >
          <ul
            className="navbar-nav ms-auto"
          >
            <li>
              <span
                className="nav-item nav-link"
                style={
                  Object {
                    "color": "cyan",
                  }
                }
              >
                Fernando
              </span>
            </li>
            <li>
              <button
                className="nav-item nav-link btn btn-outline-info"
                onClick={[Function]}
              >
                Logout
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      className="container mt-4"
    >
      <div>
        <h1>
          MarvelComponent
        </h1>
        <hr />
        <div
          className="row animate__animated animate__fadeIn"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Sort
          </button>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Spider Man"
                className="card-img-top"
                src="./assets/heroes/marvel-spider.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Captain America"
                className="card-img-top"
                src="./assets/heroes/marvel-captain.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Iron Man"
                className="card-img-top"
                src="./assets/heroes/marvel-iron.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Thor"
                className="card-img-top"
                src="./assets/heroes/marvel-thor.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Hulk"
                className="card-img-top"
                src="./assets/heroes/marvel-hulk.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Wolverine"
                className="card-img-top"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Daredevil"
                className="card-img-top"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Hawkeye"
                className="card-img-top"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Cyclops"
                className="card-img-top"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-sm-4 my-3"
          >
            <div
              className="card"
            >
              <img
                alt="Silver Surfer"
                className="card-img-top"
                src="./assets/heroes/marvel-silver.jpg"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  className="card-text"
                >
                  Marvel Comics
                </p>
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  More Info
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <AppRouter> Debe mostrar login si no est√° autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginComponent
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
